src/specification/inverse.ts

```diff
@@ -290,9 +398,69 @@ function simplifyMatch(match: Match, given: string): Match | null {
     };
 }
 
-function expectsSuccessor(condition: Condition, given: string) {
-    return condition.type === "path" &&
-        condition.labelRight === given &&
-        condition.rolesRight.length === 0 &&
-        condition.rolesLeft.length > 0;
+function expectsSuccessor(condition: Condition, given: string) {
+    return condition.type === "path" &&
+        condition.labelRight === given &&
+        condition.rolesRight.length === 0 &&
+        condition.rolesLeft.length > 0;
+}
+
+/**
+ * Validates that a specification meets the required invariant properties:
+ * 1. All matches except the first must have at least one path condition
+ * 2. Path conditions must reference labels that appear earlier in the specification
+ * 
+ * @param specification The specification to validate
+ * @returns true if the specification is valid
+ * @throws Error with descriptive message if the specification violates invariants
+ */
+export function validateSpecificationInvariant(specification: Specification): boolean {
```

Comment:
Be sure to recursively validate the existential conditions of each given and of each match.